### PR TITLE
[SSD-2897] Cross-domain user identification package

### PR DESCRIPTION
**Pull request by maurimtl**
# Cross-Domain User Identification for Tracking Pixels

## Overview
This PR introduces `@workleap/user-identification`, a new package that enables cross-domain user identification for marketing and analytics tracking pixels across the Workleap ecosystem.

## Problem Statement
- **Marketing websites** (Webflow) load tracking pixels without user context, causing all users to appear as "new users"
- **Multiple tracking platforms** (Common Room, Meta/Facebook, LinkedIn, Google Analytics, etc.) need access to comprehensive user traits
- **Cross-domain challenges** prevent sharing user identification between `*.workleap.com` subdomains
- **No centralized solution** for user trait management across marketing and product domains

## Solution
A comprehensive user identification package that:
- ✅ **Extends existing `.workleap.com` cookie system** from `@workleap/telemetry`
- ✅ **Dual persistence strategy** (cookies for cross-domain + localStorage for performance)
- ✅ **Comprehensive user traits** based on LogRocket's implementation
- ✅ **TypeScript-first** with complete type safety
- ✅ **Production-ready** with defensive programming and size warnings
- ✅ **Framework agnostic** - works with React, vanilla JS, any tracking pixel

### 🎯 Core API
```typescript
// Identify user (called in product apps after login)
identify(userId, {
    organizationId: "org-123",
    organizationName: "Acme Corp", 
    email: "user@acme.com",
    name: "John Doe",
    isAdmin: true,
    // ... comprehensive traits
});

// Read user data (marketing website, tracking pixels)
const user = getCurrentUser();
if (user) {
    // Available for Common Room, Meta, LinkedIn, etc.
    window.signals?.identify?.(user.userId, user.traits);
    fbq('identify', user.userId, user.traits);
    gtag('config', 'GA_ID', { user_id: user.userId });
}
```

### 🌐 Cross-Domain Persistence
- **Cookies**: `.workleap.com` domain for cross-subdomain access
- **localStorage**: Same-origin performance optimization
- **Size monitoring**: Warns when approaching 4KB cookie limits
- **SSR safe**: Graceful degradation in server environments

### 📊 Comprehensive User Traits
Based on LogRocket's proven implementation:
- Core identity (`userId`, `organizationId`, `email`, `name`)
- Role-based permissions (`isAdmin`, `isExecutive`, `isCollaborator`)
- Product-specific data (`planCode`, `isReportingManager`)
- Profile information (`jobTitle`, `department`, `hireDate`)

### Product Apps (Login/Session Restoration)
```typescript
import { identify } from "@workleap/user-identification";

// After successful authentication
identify(user.id, user.traits);
```

### Marketing Website (No Installation Required)
```javascript
// Read directly from localStorage/cookies
const userData = localStorage.getItem("wl-user-traits");
if (userData) {
    const user = JSON.parse(userData);
    // Use with any tracking pixel
    window.signals?.identify?.(user.userId, user.traits);
}
```

### 🏗️ Package Structure
- **Follows established patterns** from `@workleap/honeycomb`, `@workleap/logrocket`
- **Modular design** with reusable `WorkleapStorage` utilities
- **Type-safe exports** with comprehensive TypeScript interfaces
- **Tree-shakeable** ESM build with optimized bundle size

### 🧪 Testing
- **18 comprehensive tests** covering all API functions
- **Storage simulation** accurately mimicking browser behavior
- **Bug validation** tests for edge cases and error scenarios
- **100% passing** test suite with robust mocking

### 📖 Documentation
- **Complete API reference** with examples for each function
- **Integration guides** for React apps, marketing websites
- **Migration examples** showing real-world usage patterns
- **Storage strategy documentation** explaining cross-domain persistence

## Files Changed
- **New Package**: `packages/user-identification/` (complete package)
- **Root README**: Added to packages table
- **Documentation**: `docs/user-identification/` (complete docs)
- **Integration Guide**: Updated `docs/introduction/getting-started.md`

## Validation
- ✅ **Build**: `pnpm build` passes
- ✅ **Tests**: All 18 tests pass
- ✅ **Linting**: ESLint configuration added, no errors
- ✅ **Bundle**: Optimized 24.4 kB total (6.3 kB gzipped)
- ✅ **Integration**: Tested with storage simulation

## Next Steps
1. **Review and merge** this PR
2. **Publish package** to npm (`@workleap/user-identification@1.0.0`)
3. **Product integration** - add `identify()` calls to login flows
4. **Marketing integration** - update Webflow tracking pixels
5. **Monitoring** - track user identification rates across domains

---

**Ready for review!** This package provides a robust, production-ready solution for cross-domain user identification that will enhance all tracking pixel implementations across Workleap's marketing and product ecosystem.

### CODE REVIEW

**Review by lilmano18 — COMMENTED** [review]
The current implementation looks solid, but could we add error handling for storage failures?

For example:

    - localStorage quota exceeded
    - Cookie setting failures (disabled cookies)
    - Invalid user data scenarios

**Review by lilmano18 — APPROVED** [review]

File: packages/user-identification/src/identify.ts [hunk 1]
```diff
@@ -0,0 +1,171 @@
+import { __clearUserContext, createUserContext, getUserContext, type UserContext } from "./UserContext.js";
+import { createUserTraitsStorage } from "./storageUtils.js";
```
Comment (alexasselin008) [inline packages/user-identification/src/identify.ts:2]: in all your files, the wrong extension is used. It should always be .ts not .js

File: docs/user-identification/reference/getCurrentUser.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+---
+
+# getCurrentUser
+
+Retrieves the current user context from storage. This function reads persisted user identification data from cookies and localStorage.
+
+## Reference
+
+```ts
+const userContext = getCurrentUser(options?)
```
[the remark this answers is not in the document]
Comment (alexasselin008) [reply docs/user-identification/reference/getCurrentUser.md:17]: make sure to update your other reference/*.md files

File: docs/user-identification/samples.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+---
+order: 0
+label: Samples
+meta:
+    title: Samples - User Identification
+---
+
+# Samples
+
+- :icon-mark-github: [All platforms sample](https://github.com/workleap/wl-telemetry/tree/main/samples/all-platforms)
```
Comment (alexasselin008) [inline docs/user-identification/samples.md:10]: This documentation article is odd. The samples are not using `@workleap/user-identification`.

File: packages/user-identification/tests/identify.test.ts [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+    });
+
+    afterEach(() => {
+        vi.clearAllMocks();
+        __clearUserContext();
+        storageData.clear(); // Clear the simulated storage
+    });
+
+    describe("identify", () => {
+        test("should identify a user and return UserContext", () => {
```
Comment (alexasselin008) [inline packages/user-identification/tests/identify.test.ts:101]: The test names don't really follow the pattern of this repo

File: packages/user-identification/src/userTypes.ts [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+    hireDate?: string;
+    role?: string;
+    plan?: string;
+}
+
+/**
+ * User traits stored in cookies and localStorage for cross-domain persistence.
+ * Uses normalized property names for consistent access across platforms.
+ */
+export interface StoredUserTraits extends Record<string, unknown> {
```
[the remark this answers is not in the document]
Comment (alexasselin008) [reply packages/user-identification/src/userTypes.ts:63]: taken from logrocket

File: packages/user-identification/src/identify.ts (hunk 1 above)
Comment (maurimtl) [reply packages/user-identification/src/identify.ts:2]: fixed this for the new submission

File: docs/user-identification/reference/getCurrentUser.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (maurimtl) [reply docs/user-identification/reference/getCurrentUser.md:17]: fixed this for the new subsmission

File: docs/user-identification/samples.md (hunk 3 above)
Comment (maurimtl) [reply docs/user-identification/samples.md:10]: agree, removed

File: packages/user-identification/src/userTypes.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (maurimtl) [reply packages/user-identification/src/userTypes.ts:63]: Addressed in the next submission

```
export interface StoredUserTraits {
    "User Id": string;
    "Organization Id": string;
    // ... rest of properties
}
```

1. Doesn't extend Record<string, unknown>
2. Uses "User Id" (not just "User") as the reviewer suggested
3. Has specific, typed properties instead of the error-prone generic record

File: packages/user-identification/tests/identify.test.ts (hunk 4 above)
Comment (maurimtl) [reply packages/user-identification/tests/identify.test.ts:101]: also addressed for the next one

**Comment by alexasselin008** [comment]
We will close this PR for now, and @patricklafrance will take a look at it later to see if it's something we want to include
